lib/http/response.rb

```diff
@@ -92,13 +92,18 @@ def flush
     #   (not an integer, e.g. empty string or string with non-digits).
     # @return [Integer] otherwise
     def content_length
+      # http://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.3.3
+      # Clause 3: "If a message is received with both a Transfer-Encoding
+      # and a Content-Length header field, the Transfer-Encoding overrides the Content-Length.        
+      return nil if headers[Headers::TRANSFER_ENCODING]
+
       value = @headers[Headers::CONTENT_LENGTH]
       return unless value
 
       begin
         Integer(value)
       rescue ArgumentError
-        nil
+        raise(HeaderSyntaxError, 'wrong Content-Length format')
```

Comment:
I agree. I think logic in that part should remain as is. Also I think that if Content-Length is given with wrong value we should treat it as "no content length" given. I agree that it's probably pretty rare edge case, but if it will happen I think it's developer who should decide how to handle that (read till the end or terminate) not our gem.